### PR TITLE
Limit the number of rendered overloads on signature mismatch

### LINKED ISSUE #9880: Compiler crashed: Invalid memory access (signal 11) at address 0x7f44967ea000
```shell
$ crystal build main.cr
Invalid memory access (signal 11) at address 0x7f44967ea000
[0x7f45761bdf96] ???
[0x7f4576107beb] ???
[0x7f4576f8b831] ???
```

```shell
$ uname -a
Linux LAPTOP-Q3L7M1FS 4.4.0-18362-Microsoft #476-Microsoft Fri Nov 01 16:53:00 PST 2019 x86_64 GNU/Linux

$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

$ crystal --version
Crystal 0.35.1 [5999ae29b] (2020-06-19)

LLVM: 8.0.0
Default target: x86_64-unknown-linux-gnu
```

Code:
https://gist.github.com/arloan/861b4b3daa5c4df9579a13771c50e933

### PR DESCRIPTION
**Pull request by caspiano**
Fixes #9880

Looking for feedback on a sane limit to the number of rendered overloads 👀

### CODE REVIEW

File: src/compiler/crystal/semantic/call_error.cr [hunk 1]
```diff
@@ -161,7 +164,8 @@ class Crystal::Call
           uniq_arg_names = uniq_arg_names.size == 1 ? uniq_arg_names.first : nil
           unless missing.empty?
             msg << "\nCouldn't find overloads for these types:"
-            missing.each do |missing_types|
+
+            missing.each.first(MAX_RENDERED_OVERLOADS).each do |missing_types|
```
Comment (Sija) [inline src/compiler/crystal/semantic/call_error.cr:168]:
```suggestion
            missing.first(MAX_RENDERED_OVERLOADS).each do |missing_types|
```

**Comment by HertzDevil** [comment]
I don't think this prevents `missing` itself from having millions of elements.

**Comment by caspiano** [comment]
@HertzDevil good point. I'm just wrapping my head around the matching code and then I will update this PR

File: src/compiler/crystal/semantic/call_error.cr [hunk 2]
```diff
@@ -37,6 +37,9 @@ class Crystal::Path
 end
 
 class Crystal::Call
+  # :nodoc:
+  MAX_RENDERED_OVERLOADS = 500
```
Comment (lbguilherme) [inline src/compiler/crystal/semantic/call_error.cr:41]: Most terminals have can scroll up 500 lines, this means it won't be possible to read the full error message.

I would suggest 10 or 20.

**Comment by caspiano** [comment]
@HertzDevil looks like these changes fix the issue. I think a little bit more of an involved refactor would be required for failing before exhausting the `missing` array 

The output of OP's example against these changes can be [found here](https://github.com/caspiano/crystal/runs/2981779031#step:5:91)

**Review by beta-ziliani — APPROVED** [review]
This is a sensible change, even if it won't limit the number of overloads found

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**